pkg/ipc/ipc_test.go

```diff
@@ -188,3 +188,41 @@ func TestParallel(t *testing.T) {
 		}
 	}
 }
+
+func TestZlib(t *testing.T) {
+	t.Parallel()
+	target, err := prog.GetTarget(targets.TestOS, targets.TestArch64)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg, opts, err := ipcconfig.Default(target)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg.Flags |= FlagDebug
+	cfg.Executor = buildExecutor(t, target)
+	defer os.Remove(cfg.Executor)
+	env, err := MakeEnv(cfg, 0)
+	if err != nil {
+		t.Fatalf("failed to create env: %v", err)
+	}
+	defer env.Close()
+	r := rand.New(testutil.RandSource(t))
+	for i := 0; i < 10; i++ {
```

Comment:
Do we really need 10 iterations for this test?

Reply:
The zlib code is quite complex and I am not sure even 10 iterations cover all code there. I was thinking about a bit more elaborate function that creates random "images" that would trigger more branches (e.g. repeat some chunks multiple times, add runs of 0s, etc). But I did not get to it yet.